### PR TITLE
feat: retrieve key from lambda

### PR DESCRIPTION
**Pull request by JPedroBorges**
This PR aims to retrieve the key from a lambda function instead of obfuscated code.

### CODE REVIEW

File: src/handler/analytics.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const GA_ENDPOINT = "https://www.google-analytics.com/mp/collect";
+const KEY_URL = "https://europe-west1-room-elephnat-key-retriever.cloudfunctions.net/getKeys";
```
Comment (Marianarodrigues7) [inline src/handler/analytics.js:2]: Please check the service for a typo in elephant.

File: src/handler/analytics.js [hunk 2]
```diff
@@ -75,90 +76,16 @@ async function getOrCreateSessionId() {
...
-      .join("")
-  );
+async function getKey() {
+  let { apiKey } = await chrome.storage.local.get("apiKey");
+  if (apiKey !== null && apiKey !== undefined) return apiKey;
+
+  const request = await fetch(KEY_URL, { method: "POST" });
+  const json = await request.json();
+
+  apiKey = json.googleSecreteKey;
```
Comment (Marianarodrigues7) [inline src/handler/analytics.js:86]:
```suggestion
  apiKey = json.googleSecretKey;
```

**Review by Marianarodrigues7 — APPROVED** [review]

File: src/handler/analytics.js (hunk 1 above)
Comment (JPedroBorges) [reply src/handler/analytics.js]: done.

File: src/handler/analytics.js (hunk 2 above)
Comment (JPedroBorges) [reply src/handler/analytics.js]: done.

File: src/handler/analytics.js [hunk 3]
```diff
@@ -75,90 +76,16 @@ async function getOrCreateSessionId() {
...
+  await chrome.storage.local.set({ apiKey });
```
[the remark this answers is not in the document]
Comment (JPedroBorges) [reply src/handler/analytics.js]: done.

**Pull request merged**